### ISSUE TITLE
Code Documentation/Testing

### ISSUE THREAD

**Issue by JoeBiellik** [TV-Rename/tvrename#220]
Considering this project has been around for over a decade and I don't see people's need for the functionality changing anytime soon, I think it makes a lot of sense for us to document the code for future maintainers.

I think at the very least we should focus on documenting any changes with basic code comments going forwards. [I added some basic method comments](https://github.com/JoeBiellik/tvrename/blob/a732221a7776171f93b8f41da95ac870051989ba/TVRename%23/TVRename/PathManager.cs) to the code I've refactored so far, but primarily to help with Visual Studio IntelliSense. Is it worth us formalizing documentation?

It would require a lot of work to write tests, but we could use AppVeyor to run unit tests for each PR, which would help a lot with identifying bugs.

I'd be interested in peoples thoughts.

**Comment by MarkSummerville** [comment]
I agree that we should move in this direction. Given your experience can you suggest a framework/template we should use (for both testing and documentation)? I have no experience of writing automated tests (in circa 2002 when I was last a  commercial programmer they were not as common as they are today). 

I'd worry about going though a big one-off exercise as I fear it will kill any enthusiasm in the project, rather I'd see us gradually adding to the codebase as we refactor/add code. (akin to the current logging implementation). We could use the [wiki](https://github.com/TV-Rename/tvrename/wiki) to store the guidance notes. (this is where I envisage developer documentation going - but some should be interleaved with the code).

**Comment by JoeBiellik** [comment]
I think your suggestion of using the Github wiki for general developer notes along with code usage comments would cover it.

I agree writing unit tests for everything including the UI is unrealistic for now, but I think even just testing serialization, filename processing, and IO will suffice.

NUnit is probably the most popular unit testing framework, but frankly, for our needs, the built-in MSTest will do.
